src/api/software/asymmetric/ecc/soft_ecc.c

```diff
@@ -2134,13 +2371,129 @@ int32_t soft_ecc_mod_secp384r1(const metal_scl_t *const scl,
     return (result);
 }
 
+int32_t soft_ecc_mod_secp521r1(const metal_scl_t *const scl,
```

Comment:
This is based on NIST 186-4 one, and it's the same.

Reply:
could you look at http://delta.cs.cinvestav.mx/~francisco/arith/julio.pdf ?

Reply:
Still, (a1 + a2) mod p521 is what I have done. By the way you're not looking at the last version of it.

Reply:
It's not about the algorithm, it's about the efficiency of the implementation

Reply:
Anybody is free to do another implementation, this is not the purpose of a review to change that (at least I think so).

Reply:
that's the difference between a functional implementation and an efficient implementation.
In cryptography the efficiency is critical, otherwise why implementing this optimized modular reduction ?

Reply:
What do you want me to respond to this, you just criticize without giving any hint of what you think is wrong.

Reply:
everything is ok